### DISCUSSION TITLE
ESP32 problems

### DISCUSSION THREAD

**Discussion by gerthalberg** [grame-cncm/faust#612] [Q&A]
Hi All,
I'm trying to get the smoothDelay example up and running on an ESP32 TTGO_TAUDIO . I'm using platformIO and when compiling using the ESp IDF pluging I get the following linker error:esp/esp-idf/components/freertos/port/port_common.c:129: undefined reference to `app_main'. I have tried with both an external app_main and with the app_main in the generated .cpp file.
If I try to compile using the the standard platformIO setup, the code compiles, and I can upload it to the board but the board keeps rebooting.
Has anyone tried this?

**Comment by sletz** [comment]
Possible better is to ask one the [Faust Slack channel](https://join.slack.com/t/faustaudio/shared_invite/zt-a624szlz-fL4v2DTR~ZGlI7wARryT7g) when an #esp32 channel exists.

**Comment by andy-wi** [comment]
There are a few platformIO examples [here](https://github.com/ohmic-net/pico_dsp) for both framework esp-idf and framework arduinoespressif32 if these are any help. About the rebooting, is it a HW or SW reboot? If RTCWDT_RTC_RST it may be a power supply issue, or, you could try to add to your platformio.ini file board_build.flash_mode = dio , if it is a SW_CPU_RESET maybe something in the memory configuration in menuconfig isn't configured.
